openshift/ansible/tasks/build_react.yaml

```diff
@@ -0,0 +1,20 @@
+---
+  - name: create build param template temporarily
+    template:
+      src: ./templates/build.param.j2
+      dest: './templates/build.param'
+    vars:
+      GIT_REF: 'pull/{{ PR }}/head'
+
+  - name: create build in tools namespace using head/pulls/{{ PR }} as ref 
+    shell: >
+      oc process -n "{{ tools_namespace }}" -f "../templates/react-frontend/bc.yaml" --param-file="./templates/build.param" | 
+      oc apply -n "{{ tools_namespace }}" -f -
+
+  - name: start build
+    shell: oc -n "{{ tools_namespace }}" start-build {{ react_infra_name }}-{{ version }}
+
+  - name: cleanup run (delete temp files)
```

Comment:
if you are interested, take a look at ansible k8s modules where it source jinja2 templates directly and updates objects based on state. That way there's no need to generate tmp files and clean up afterwards.

Reply:
https://docs.ansible.com/ansible/latest/modules/k8s_module.html (ops forgot to add the link..)